src/app.js

```diff
@@ -1,13 +1,260 @@
+const TRIP_FIELDS = ['name', 'category', 'weeks', 'cost', 'about', 'continent']
+
+let tripTemplate;
+
+const loadTrips = function loadTrips(trips) {
+  const tripTable = $('#tripSection');
+  tripTable.html('');
+  trips.forEach((trip) => {
+    const generatedHTML = tripTemplate(trip.attributes);
+    tripTable.append(generatedHTML);
+  });
+}
 
 $(document).ready( () => {
-  $('main').html('<h1>Hello World!</h1>');
+  tripTemplate = _.template($('#tripTemplate').html());
+  const singleTripTemplate = _.template($('#singleTripTemplate').html());
+  const tripList = new TripList();
+  tripList.fetch();
+
+  $('#singleTripSection').hide();
+  $('#reservationForm').hide();
+  $('#createTripSection').hide();
+  $('#numberFilter').hide();
+
+
+  $('#tripSection').on('click', 'tr', (e) => {
+    $('#singleTripSection').show();
+    $('#reservationForm').show();
+    const baseURL = 'https://ada-backtrek-api.herokuapp.com/trips/'
+    $.get(`${baseURL}${e.currentTarget.id}`, (response) => {
+      const generatedHTML = singleTripTemplate(response);
+      $('#singleTrip').html(generatedHTML);
+    }).fail(() => {
+      console.log('The post call failed');
+    });
+  });
+
+  tripList.on('sort', loadTrips, tripList);
+  tripList.on('update', loadTrips, tripList);
+
+  $('#createTripButton').on('click', () => {
+    $('#createTripSection').show();
+  });
+
+  $('#createTripForm').on('submit', (e) => {
+    e.preventDefault();
+    let valid = true;
+    if ($('#nameField').val() === '') {
+      $('#nameField').css('background-color','red');
+      valid = false;
+    } else {
+      $('#nameField').css('background-color','white');
+    }
+
+    if ($('#aboutField').val() === '') {
+      $('#aboutField').css('background-color','red');
+      valid = false;
+    } else {
+      $('#aboutField').css('background-color','white');
+    }
+
+    if ($('#categoryField').val() === '') {
+      $('#categoryField').css('background-color','red');
+      valid = false;
+    } else {
+      $('#categoryField').css('background-color','white');
+    }
+
+    if ($('#weeksField').val() === '') {
+      $('#weeksField').css('background-color','red');
+      valid = false;
+    } else {
+      $('#weeksField').css('background-color','white');
+    }
+
+    if ($('#costField').val() === '') {
+      $('#costField').css('background-color','red');
+      valid = false;
+    } else {
+      $('#costField').css('background-color','white');
+    }
+
+    if (valid){
+      const url = 'https://ada-backtrek-api.herokuapp.com/trips';
+      const data = $('#createTripForm').serialize();
+
+      $.post(url, data, (response) => {
+        const newTrip = new Trip(response);
+        tripList.add(newTrip);
+        $('#createTripSection').hide();
+      }).fail(() => {
+        console.log('The post call failed');
+      });
+      return false;
+    }
+  });
+
+  $('#reservationForm').on('submit', () => {
+    let valid = true;
+    if ($('#reservationName').val() === '') {
```

Comment:
These type of validations **REALLY** need to be in the model.  Add a [`validate`](http://backbonejs.org/#Model-validate) function to your model.